geoh5py/objects/surveys/electromagnetics/base.py

```diff
@@ -314,46 +332,57 @@ def edit_metadata(self, entries: dict[str, Any]):
 
         :param entries: Metadata key value pairs.
         """
+        metadata = self.metadata.copy()
         for key, value in entries.items():
             if key == "Property groups":
                 self._edit_validate_property_groups(value)
 
             elif value is None:
-                if key in self.metadata["EM Dataset"]:
-                    del self.metadata["EM Dataset"][key]
+                if key in metadata["EM Dataset"]:
+                    del metadata["EM Dataset"][key]
 
             else:
-                self.metadata["EM Dataset"][key] = value
+                metadata["EM Dataset"][key] = value
 
-        for dependent in ["receivers", "transmitters", "base_stations"]:
-            if getattr(self, dependent, None) is not None:
-                getattr(self, dependent).metadata = self.metadata
+        self.metadata = metadata
 
     def _edit_validate_property_groups(
-        self, values: PropertyGroup | list[PropertyGroup] | None
+        self, values: PropertyGroup | list[PropertyGroup] | list[str] | None
     ):
         """
         Add or append property groups to the metadata.
 
         :param value:
         """
-        if not isinstance(values, (PropertyGroup, type(None))):
-            raise TypeError(
-                "Input value for 'Property groups' must be a PropertyGroup, "
-                "list of PropertyGroup or None."
-            )
-
-        if values is None:
+        if not values:
             self.metadata["EM Dataset"]["Property groups"] = []
             return
 
         if not isinstance(values, list):
             values = [values]
 
+        groups = (
+            {group.name: group for group in self.property_groups}
+            if self.property_groups
+            else {}
+        )
+
         for value in values:
-            if self.property_groups is not None and value not in self.property_groups:
+            if self.property_groups is None:
+                continue
+
+            if not isinstance(value, (PropertyGroup, str)):
+                raise TypeError(
+                    "Input value for 'Property groups' must be a PropertyGroup, "
```

Comment:
**issue:** exception message is inconsistent with the types used inthe `isinstance()` test

Reply:
Done. Replaced with base_copy